include/roboteam_ai/treeinterp/TreeProtoType.h

```diff
@@ -0,0 +1,33 @@
+//
+// Created by jesse on 14.10.19.
+//
+
+#include <include/roboteam_ai/bt/tactics/DefaultTactic.h>
+#include "bt/BehaviorTree.hpp"
+#include "bt/Role.h"
+#ifndef RTT_MAKE_TREE_WITHOUT_JSONS_H
+#define RTT_MAKE_TREE_WITHOUT_JSONS_H
+
+#endif //RTT_MAKE_TREE_WITHOUT_JSONS_H
+
+
+namespace bt {
+
+class TreeProtoType {
+ public:
+    std::shared_ptr<bt::DefaultTactic> createDefensiveTactic(std::shared_ptr<Blackboard> bb);
+    std::shared_ptr<bt::BehaviorTree> createNormalPlayStrategy();
+    std::shared_ptr<bt::Role> createDefenderRole(std::string name);
+
+private:
+    std::vector<std::pair<std::string, rtt::ai::robotDealer::RobotType>> robots = {
```

Comment:
I am generally not a fan of long default arguments, because you would expect this information in the constructor. When you run in release mode the overhead is optimised away anyways.

Reply:
Also, it keeps the headers small and readable

Reply:
Putting this in the constructor is a good idea! I think now there is a separate file for each of these robot vectors, but I feel like keeping them with the tactic/tree makes sense now that the jsons are out of the picture. The constructor does seem like a better place to do this than the header file :)